stats/opentelemetry/client_metrics.go

```diff
@@ -0,0 +1,262 @@
+			logger.Errorf("failed to register metric \"grpc.client.attempt.started\", will not record")
+		} else {
+			registeredMetrics.clientAttemptDuration = cad
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.sent_total_compressed_message_size"]; ok {
+		cas, err := meter.Int64Histogram("grpc.client.attempt.sent_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) sent across all request messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.attempt.sent_total_compressed_message_size\", will not record")
+		} else {
+			registeredMetrics.clientAttemptSentTotalCompressedMessageSize = cas
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.attempt.rcvd_total_compressed_message_size"]; ok {
+		car, err := meter.Int64Histogram("grpc.client.attempt.rcvd_total_compressed_message_size", metric.WithUnit("By"), metric.WithDescription("Total bytes (compressed but not encrypted) received across all response messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes."), metric.WithExplicitBucketBoundaries(DefaultSizeBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.rcvd.sent_total_compressed_message_size\", will not record")
+		} else {
+			registeredMetrics.clientAttemptRcvdTotalCompressedMessageSize = car
+		}
+	}
+
+	if _, ok := setOfMetrics["grpc.client.call.duration"]; ok {
+		ccs, err := meter.Float64Histogram("grpc.client.call.duration", metric.WithUnit("s"), metric.WithDescription("This metric aims to measure the end-to-end time the gRPC library takes to complete an RPC from the application’s perspective."), metric.WithExplicitBucketBoundaries(DefaultLatencyBounds...))
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.call.duration\", will not record")
+		} else {
+			registeredMetrics.clientCallDuration = ccs
+		}
+	}
+	csh.registeredMetrics = registeredMetrics
+}
+
+func (csh *clientStatsHandler) unaryInterceptor(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	ci := &callInfo{
+		target: csh.determineTarget(cc),
+		method: csh.determineMethod(method, opts...),
+	}
+	ctx = setCallInfo(ctx, ci)
+
+	startTime := time.Now()
+	err := invoker(ctx, method, req, reply, cc, opts...)
+	csh.perCallMetrics(ctx, err, startTime, ci)
+	return err
+}
+
+// determineTarget determines the target to record attributes with. This will be
+// "other" if target filter is set and specifies, the target name as is
+// otherwise.
+func (csh *clientStatsHandler) determineTarget(cc *grpc.ClientConn) string {
+	target := cc.CanonicalTarget()
+	if csh.mo.TargetAttributeFilter != nil {
+		if !csh.mo.TargetAttributeFilter(target) {
```

Comment:
Optional: combine these two ifs.  You can even do:

```go
if f := csh.mo.TargetAttributeFilter; f != nil && !f(target) {
}
```

Reply:
Ah, nice. Switched.